File: intrinsic/square_ex.f90
program square_ex

  use mathiness, only: square
  implicit none

  integer, parameter :: x = 5
  integer :: r

  r = square(x)
  
  write(*,*) "Square example"
  write(*,*) "A number: ", x
  write(*,*) "Its square: ", r

end program square_ex

File: intrinsic/square_ex.py
"""An example of calling a Fortran function through f2py.

Use `f2py` to build a Fortran shared object that can be called from
Python::

    f2py -c mathiness.f90 -m mathiness
 
"""
from mathiness import mathiness


x = 5
r = mathiness.square(x)

print("Square example")
print("A number:", x)
print("Its square:", r)
